File: APP_Java/app/src/main/java/com/example/sunnygym/ViewClubsActivity.java
package com.example.sunnygym;

import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.PopupMenu;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

public class ViewClubsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_view_clubs);

        List<Activity> activities = new ArrayList<>();
        activities.add(new Activity("Meridiana", R.drawable.meridiana));
        activities.add(new Activity("Can Dragó", R.drawable.candrago));
        activities.add(new Activity("Diagonal", R.drawable.diagonal));

        // Configurar RecyclerView
        RecyclerView recyclerView = findViewById(R.id.recyclerViewClubs);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setAdapter(new ActivitiesAdapter(this, activities));

    }

    /**
     * Método para mostrar el menú emergente
     *
     * @param anchorView Vista desde la cual se despliega el menú
     */
    private void showPopupMenu(View anchorView) {
        PopupMenu popupMenu = new PopupMenu(ViewClubsActivity.this, anchorView);
        popupMenu.getMenuInflater().inflate(R.menu.profile_menu, popupMenu.getMenu());

        popupMenu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
            @Override
            public boolean onMenuItemClick(MenuItem item) {
                int itemId = item.getItemId();
                if (itemId == R.id.home) {
                    startActivity(new Intent(ViewClubsActivity.this, UserProfileActivity.class));
                    return true;
                } else if (itemId == R.id.my_profile) {
                    startActivity(new Intent(ViewClubsActivity.this, MyProfileActivity.class));
                    return true;
                } else if (itemId == R.id.guided_activities) {
                    return true;
                } else if (itemId == R.id.view_clubs) {
                    return true;
                } else if (itemId == R.id.support) {
                    startActivity(new Intent(ViewClubsActivity.this, SupportActivity.class));
                    return true;
                } else if (itemId == R.id.logout) {
                    startActivity(new Intent(ViewClubsActivity.this, MainActivity.class));
                    finish();
                    return true;
                }
                return false;
            }
        });

        popupMenu.show();


    }
    // Crear lista de actividades

}

File: APP_Kotlin/app/src/main/java/com/example/sunnygym/ViewClubsActivity.kt
package com.example.sunnygym

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.PopupMenu
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class ViewClubsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_view_clubs)

        val clubsList: MutableList<Clubs> = ArrayList()
        clubsList.add(Clubs("Meridiana", R.drawable.meridiana))
        clubsList.add(Clubs("Can Dragó", R.drawable.candrago))
        clubsList.add(Clubs("Diagonal", R.drawable.diagonal))

        // Configurar RecyclerView
        val recyclerView = findViewById<RecyclerView>(R.id.recyclerViewClubs)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = ClubsAdapter(this, clubsList)
    }

    /**
     * Método para mostrar el menú emergente
     *
     * @param anchorView Vista desde la cual se despliega el menú
     */
    private fun showPopupMenu(anchorView: View) {
        val popupMenu = PopupMenu(this@ViewClubsActivity, anchorView)
        popupMenu.menuInflater.inflate(R.menu.profile_menu, popupMenu.menu)

        popupMenu.setOnMenuItemClickListener(PopupMenu.OnMenuItemClickListener { item ->
            val itemId = item.itemId
            if (itemId == R.id.home) {
                startActivity(Intent(this@ViewClubsActivity, UserProfileActivity::class.java))
                return@OnMenuItemClickListener true
            } else if (itemId == R.id.my_profile) {
                startActivity(Intent(this@ViewClubsActivity, MyProfileActivity::class.java))
                return@OnMenuItemClickListener true
            } else if (itemId == R.id.guided_activities) {
                return@OnMenuItemClickListener true
            } else if (itemId == R.id.view_clubs) {
                return@OnMenuItemClickListener true
            } else if (itemId == R.id.support) {
                startActivity(Intent(this@ViewClubsActivity, SupportActivity::class.java))
                return@OnMenuItemClickListener true
            } else if (itemId == R.id.logout) {
                startActivity(Intent(this@ViewClubsActivity, MainActivity::class.java))
                finish()
                return@OnMenuItemClickListener true
            }
            false
        })

        popupMenu.show()
    } // Crear lista de actividades
}
